pkg/client.go

```diff
@@ -22,17 +24,47 @@ func NewApiClient(optionFuncs ...ClientOptionFunc) (*Client, error) {
 	rateLimitHandler := handlers.NewRateLimitHandler()
 	middlewares := options.Middleware
 	middlewares = append(middlewares, rateLimitHandler)
-	netHttpClient := kiotaHttp.GetDefaultClient(middlewares...)
+	defaultTransport := kiotaHttp.GetDefaultTransport()
+	netHttpClient := &http.Client{
+		Transport: defaultTransport,
+	}
+
 	if options.RequestTimeout != 0 {
 		netHttpClient.Timeout = options.RequestTimeout
 	}
 
+	// Configure GitHub App authentication if required fields are provided
+	if (options.GitHubAppID != 0 || options.GitHubAppClientID != "") && options.GitHubAppInstallationID != 0 && options.GitHubAppPemFilePath != "" {
+		existingTransport := netHttpClient.Transport
+
+		if options.GitHubAppClientID != "" {
+			appTransport, err := ghinstallation.NewKeyFromFile(existingTransport, options.GitHubAppClientID, options.GitHubAppInstallationID, options.GitHubAppPemFilePath)
+			if err != nil {
+				return nil, fmt.Errorf("failed to create transport from GitHub App using clientID: %v", err)
+			}
+			netHttpClient.Transport = appTransport
+		} else {
+			appTransport, err := ghinstallation.NewKeyFromFileWithAppID(existingTransport, options.GitHubAppID, options.GitHubAppInstallationID, options.GitHubAppPemFilePath)
+			if err != nil {
+				return nil, fmt.Errorf("failed to create transport from GitHub App using appID: %v", err)
+			}
+			netHttpClient.Transport = appTransport
+		}
+	}
```

Comment:
You might consider something like below to reduce cyclomatic complexity and nesting.  Just a nit, but I found myself reading and rereading, trying to execute each code path  - what I propose might break some idiom in Go btw.. just a thought.

```suggestion
  if (options.GitHubAppID != 0 || options.GitHubAppClientID != "") && options.GitHubAppInstallationID != 0 && options.GitHubAppPemFilePath != "" {
      existingTransport := netHttpClient.Transport
      var appTransport ghinstallation.Transport
      var err error
  
      if options.GitHubAppClientID != "" {
          appTransport, err = ghinstallation.NewKeyFromFile(existingTransport, options.GitHubAppClientID, options.GitHubAppInstallationID, options.GitHubAppPemFilePath)
      } else {
          appTransport, err = ghinstallation.NewKeyFromFileWithAppID(existingTransport, options.GitHubAppID, options.GitHubAppInstallationID, options.GitHubAppPemFilePath)
      }
  
      if err != nil {
          return nil, fmt.Errorf("failed to create transport from GitHub App: %v", err)
      }
  
      netHttpClient.Transport = appTransport
  }
```